core/src/main/scala/org/scalarelational/column/property/NotNull.scala

```diff
@@ -1,8 +0,0 @@
-package org.scalarelational.column.property
-
-/**
- * @author Matt Hicks <matt@outr.com>
- */
-object NotNull extends ColumnProperty {
```

Comment:
I understand your objections, however there are several benefits that I would like to raise:

This feature makes you think whether values in a column may be optional. If so, this needs to be made explicit. While porting one large code-base to ScalaRelational, I found many columns where ``NotNull`` was specified incorrectly. There were also a couple of mismatches where column values could have been ``null``, but the type in the mapped ``case class`` was not an ``Option``. Such errors are now detected early, i.e. before going into production and potentially causing harm.

We also don't need to define valid type conversions anymore (the ``EnhancedMethod.convertTo()`` call was replaced by ``asInstanceOf``). In my [options branch](https://github.com/tindzk/scalarelational/commits/options), I needed to introduce workarounds in several places to get ``T`` converted to ``Option[T]``. Now, the whole logic is centralised in [one function](https://github.com/tindzk/scalarelational/blob/strict-conversions/core/src/main/scala/org/scalarelational/datatype/DataType.scala#L21). The solution is cleaner and less-error prone. It had positive ramifications on other parts of ScalaRelational, for example all ``null`` checks in ``DataType[_]`` became redundant (see [EnumDataType](https://github.com/outr/scalarelational/pull/16/files#diff-abd71fcb2dab5663c3d73e5bd27ea884R129)).

I agree with you that we should be very cautious when introducing new features, especially when they shift the direction. In this particular case, I'm convinced that the positive outweighs the negative aspects.